crates/burn-import/src/onnx/op_configuration.rs

```diff
@@ -1283,60 +1283,74 @@ pub fn shape_config(curr: &Node) -> (usize, usize) {
+        // If the input is not provided, return an empty vector
+        if node.inputs.get(index).is_none() {
+            return Vec::new();
+        }
+
         match &node.inputs[index].value {
-            Some(Data::Int64s(shape)) => shape
-                .iter()
-                .enumerate()
-                .map(|(i, x)| {
-                    if x.is_negative() {
-                        tensor_shape[i] - x.wrapping_abs() as usize
-                    } else {
-                        *x as usize
-                    }
-                })
-                .collect(),
+            Some(Data::Int64s(shape)) => shape.clone(),
+
             _ => panic!("Tensor data type must be int64"),
         }
     }
 
-    ensure_1d_tensor(node, 1);
-    ensure_1d_tensor(node, 2);
-
-    let starts = get_input_values(node, 1);
-    let ends = get_input_values(node, 2);
+    let mut starts = get_input_values(node, 1);
+    let mut ends = get_input_values(node, 2);
+    let mut axes = get_input_values(node, 3);
+    let mut steps = get_input_values(node, 4);
 
+    // https://burn.dev/docs/burn/prelude/struct.Tensor.html#method.slice
+    // TODO default missing axes ranges to the full range of the corresponding axis
     for (key, value) in node.attrs.iter() {
         match key.as_str() {
-            "axes" => {
-                let mut i = 0;
-                value.clone().into_i64s().iter().for_each(|x| {
-                    assert_eq!(*x, i, "Slice: axes must be consecutive");
-                    i += 1;
-                })
-            }
-            "steps" => value.clone().into_i64s().into_iter().for_each(|x| {
-                if x != 1 {
-                    panic!("Slice: steps other than 1 are not supported");
-                }
-            }),
+            "starts" => starts = value.clone().into_i64s(),
+            "ends" => ends = value.clone().into_i64s(),
+            "axes" => axes = value.clone().into_i64s(),
+            "steps" => steps = value.clone().into_i64s(),
             _ => {}
         }
     }
 
-    (starts, ends)
+    if !steps.is_empty() && steps.iter().any(|&x| x != 1) {
+        panic!("Slice: steps other than 1 are not supported");
+    }
+
+    // Extract the shape of the input tensor
+    let input_dim = match node.inputs.first().unwrap().clone().ty {
+        ArgType::Tensor(tensor) => tensor.dim,
+        _ => panic!("Only tensor input is valid"),
+    };
+
+    // If axes is not provided, it defaults to all axes
+    if axes.is_empty() {
+        axes = (0..starts.len() as i64).collect();
+    }
+
+    // assert len(starts) == len(ends) == len(axes)
+    if starts.len() != ends.len() || starts.len() != axes.len() {
+        panic!("Slice: starts, ends, and axes must have the same length");
+    }
+
+    // If dim is negative, it is counted from the end
+    // Negative value means counting dimensions from the back.
+    for i in 0..axes.len() {
```

Comment:
Didn't notice it before, and somehow clippy didn't run with the CI, but in the review mode it's getting triggered:

```
[clippy] reported by reviewdog 🐶
error: the loop variable `i` is only used to index `axes`
    --> crates/burn-import/src/onnx/op_configuration.rs:1339:14
     |
1339 |     for i in 0..axes.len() {
     |              ^^^^^^^^^^^^^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
     = note: `-D clippy::needless-range-loop` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(clippy::needless_range_loop)]`
help: consider using an iterator
     |
1339 |     for <item> in &mut axes {
     |         ~~~~~~    ~~~~~~~~~
```

Fair point clippy 😄

Reply:
My local run-checks is broken =(

Fixed.